marlowe-playground-client/src/Halogen/Monaco.purs

```diff
@@ -155,3 +196,25 @@ handleQuery (SetModelMarkers markers next) = do
       model <- Monaco.getModel editor
       Monaco.setModelMarkers monaco model owner markers
       pure next
+
+handleQuery (SetKeyBindings DefaultBindings next) =
+  withEditor \editor -> do
+    { deactivateBindings } <- get
+    liftEffect deactivateBindings
+    pure next
+
+handleQuery (SetKeyBindings Emacs next) =
+  withEditor \editor -> do
+    { deactivateBindings } <- get
+    liftEffect deactivateBindings
+    disableEmacsMode <- liftEffect $ Monaco.useEmacsBindings editor
+    modify_ (\s -> s { deactivateBindings = disableEmacsMode })
+    pure next
+
+handleQuery (SetKeyBindings Vim next) =
+  withEditor \editor -> do
+    { deactivateBindings } <- get
+    liftEffect deactivateBindings
+    disableVimMode <- liftEffect $ Monaco.useVimBindings editor
```

Comment:
Being picky, but `useXBindings` should probably be `enableXBindings`.

Reply:
😒 
Done.